server/src/main/java/com/cloud/api/ApiResponseHelper.java

```diff
@@ -621,11 +622,14 @@ public VMSnapshotResponse createVMSnapshotResponse(VMSnapshot vmSnapshot) {
         vmSnapshotResponse.setDisplayName(vmSnapshot.getDisplayName());
         UserVm vm = ApiDBUtils.findUserVmById(vmSnapshot.getVmId());
         if (vm != null) {
-            vmSnapshotResponse.setVirtualMachineid(vm.getUuid());
+            vmSnapshotResponse.setVirtualMachineId(vm.getUuid());
+            vmSnapshotResponse.setVirtualMachineName(Strings.isNullOrEmpty(vm.getDisplayName()) ? vm.getHostName() : vm.getDisplayName());
+            vmSnapshotResponse.setVirtualMachineName(vm.getHostName());
```

Comment:
Why is setVirtualMachineName() added twice @shwstppr ?

Reply:
@rhtyd that was by mistake in my previous commit :facepalm: . Fixed it